mozci/push.py

```diff
@@ -106,20 +106,37 @@ def tasks(self):
             list: A list of `Task` objects.
         """
         args = Namespace(rev=self.rev, branch=self.branch)
-        data = run_query('push_results', args)['data']
-
-        tasks = []
-        for kwargs in data:
-            # Do a bit of data sanitization.
-            if any(a not in kwargs for a in ('label', 'duration', 'result', 'classification')):
-                continue
-
-            if kwargs['duration'] <= 0:
+        tasks = defaultdict(dict)
+
+        def add(data):
+            for task in data:
+                if 'id' not in task:
+                    continue
+                tasks[task['id']].update(task)
+
+        # Gather information from the treeherder and task tables.
+        for table in ('treeherder', 'task', 'unittest'):
+            add(run_query('push_tasks_from_{}'.format(table), args)['data'])
+
+        # Normalize tags.
+        tasks = tasks.values()
+        for task in tasks:
+            if not task.get('tags'):
                 continue
-
-            tasks.append(Task(**kwargs))
-
-        return tasks
+            task['tags'] = {t['name']: t['value'] for t in task['tags']}
+
+        # If we are missing one of these keys, discard the task.
```

Comment:
Same here, should we log a warning when we discard a task or is it too frequent?